src/server_manager/web_app/digitalocean_app/create_server_app.ts

```diff
@@ -38,58 +37,50 @@ const DIGITALOCEAN_FLAG_MAPPING: {[cityId: string]: string} = {
 
 @customElement('digitalocean-create-server-app')
 export class DigitalOceanCreateServerApp extends LitElement {
-  // External events
-  public static EVENT_SERVER_CREATE_STARTED = 'DigitalOceanCreateServerApp#ServerCreateStarted';
-  public static EVENT_SERVER_CREATE_FAILED = 'DigitalOceanCreateServerApp#ServerCreateFailed';
-  public static EVENT_AUTHORIZATION_ERROR = 'DigitalOceanCreateServerApp#AuthorizationError';
-  // Internal events
-  private static EVENT_CREATE_SERVER_REQUESTED = 'DigitalOceanCreateServerApp#_RequestCreateServer';
+  public static EVENT_SERVER_CREATED = 'do-create-server-app-server-created';
+  public static EVENT_AUTHORIZATION_ERROR = 'do-create-server-app-authorization-error';
 
   @property({type: Function}) localize: Function;
 
+  private digitalOceanServerRepository: ManagedServerRepository;
   private notificationManager: OutlineNotificationManager;
-  private eventEmitter = new EventEmitter();
-
-  constructor() {
-    super();
-    this.addEventListener(OutlineRegionPicker.EVENT_REGION_SELECTED, (event: CustomEvent) => {
-      this.eventEmitter.emit(
-          DigitalOceanCreateServerApp.EVENT_CREATE_SERVER_REQUESTED, event.detail.regionId);
-    });
-  }
 
   render() {
-    return html`<outline-region-picker-step .localize=${
-        this.localize}></outline-region-picker-step>`;
+    return html`<outline-region-picker-step .localize=${this.localize} 
+        @outline-region-picker-region-selected="${
+        this.onRegionSelected}"></outline-region-picker-step>`;
+  }
+
+  async onRegionSelected(event: CustomEvent) {
+    const regionPicker =
+        this.shadowRoot.querySelector('outline-region-picker-step') as OutlineRegionPicker;
+    regionPicker.isServerBeingCreated = true;
+
+    try {
+      const server =
+          await this.createServer(this.digitalOceanServerRepository, event.detail.regionId);
+      const serverCreatedEvent =
+          makePublicEvent(DigitalOceanCreateServerApp.EVENT_SERVER_CREATED, {server});
+      this.dispatchEvent(serverCreatedEvent);
+    } catch (error) {
+      this.notificationManager.showError('error-server-creation');
+    }
   }
```

Comment:
I think we should set `regionPicker.isServerBeingCreated = false` in a finally block here